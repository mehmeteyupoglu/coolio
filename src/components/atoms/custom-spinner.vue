<template>
  <v-progress-circular
    v-if="loading"
    :width="1"
    color="primary"
  ></v-progress-circular>
</template>

<script>
/**
 * A @component that appears on loading
 * accepts a loading state
 * @example <custom-spinner :loading="loading"></custom-spinner>
 */

export default {
  name: "CustomSpinner",
  props: {
    loading: {
      type: Boolean,
      default: false,
    },
  },
  data: () => ({}),
};
</script>
