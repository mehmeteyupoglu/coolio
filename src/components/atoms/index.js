import SearchInput from "./search-input.vue";
import CustomSpinner from "./custom-spinner.vue";

export { SearchInput, CustomSpinner };
