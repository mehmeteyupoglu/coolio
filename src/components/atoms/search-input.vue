<template>
  <div>
    <v-row justify="center">
      <v-col cols="12" sm="6" md="3">
        <v-text-field
          v-model="title"
          label="Enter a keyword such as awesome"
          solo
          clearable
          filled
        ></v-text-field>
      </v-col>
    </v-row>
  </div>
</template>

<script>
/**
 * Streamline solution for handling async search activities
 * @component
 * @example
 * <search-input />
 */

export default {
  name: "SearchInput",

  data: () => ({}),
  computed: {
    title: {
      set(value) {
        this.$store.commit("setTitle", value);
      },
      get() {
        return this.$store.getters.getTitle;
      },
    },
  },
  watch: {
    title() {
      this.$store.dispatch("loadProducts");
    },
  },
};
</script>
