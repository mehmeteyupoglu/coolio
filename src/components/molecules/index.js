import CustomBtnGroup from "./custom-btn-group.vue";
import LogoGroup from "./logo-group.vue";
import CustomSnackbar from "./custom-snackbar.vue";
import RenderChips from "./render-chips";

export { CustomBtnGroup, LogoGroup, CustomSnackbar, RenderChips };
