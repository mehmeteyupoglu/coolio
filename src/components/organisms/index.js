import CustomAppBar from "./app-bar.vue";

export { CustomAppBar };
