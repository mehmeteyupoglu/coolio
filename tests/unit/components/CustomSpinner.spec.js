import { mount } from "@vue/test-utils";
import { CustomSpinner } from "../../../src/components/atoms";

describe("CustomSpinner", () => {
  it("should render to a snapshot when loading state is false", () => {
    const wrapper = mount(CustomSpinner, {
      propsData: {
        loading: false,
      },
    });
    expect(wrapper.props().loading).toBe(false);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should render to a snapshot when loading state is true", () => {
    const wrapper = mount(CustomSpinner, {
      propsData: {
        loading: true,
      },
    });
    expect(wrapper.props().loading).toBe(true);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should render to a snapshot when loading state is null", () => {
    const wrapper = mount(CustomSpinner, {
      propsData: {
        loading: null,
      },
    });
    expect(wrapper.props().loading).toBe(null);
    expect(wrapper.element).toMatchSnapshot();
  });
});
