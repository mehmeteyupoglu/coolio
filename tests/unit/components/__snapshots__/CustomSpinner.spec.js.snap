// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSpinner should render to a snapshot when loading state is false 1`] = `<!---->`;

exports[`CustomSpinner should render to a snapshot when loading state is null 1`] = `<!---->`;

exports[`CustomSpinner should render to a snapshot when loading state is true 1`] = `
<v-progress-circular
  color="primary"
  width="1"
/>
`;
